conformance/lang/expressions/mapping_constructor_expr.balt

```diff
@@ -0,0 +1,1917 @@
+
+    map<any> _ = {fieldA: 545, a, "field field": "monitor", "fieldB": true, "field field": 544}; // @error Duplicate key `field field`
+
+    map<any> _ = {field\u{0043}: 454, "field\u{0043}": false}; // @error Duplicate key `fieldC`
+    
+    map<any> _ = {a\\: 454, "a\\": false}; // @error Duplicate key `a\`
+}
+
+Test-Case: output
+Description: Test usage of spread field in mapping constructor using closed record types.
+Labels: any, boolean, error-type, error-constructor-expr, mapping-constructor-expr, map-type
+       record-type, spread-field, string, union-type
+
+function init() {
+    record {|int fieldA1; boolean fieldA2;|} a = {fieldA1: 4534, fieldA2: false};
+
+    printResult({field1: true, field2: 4345, ...a}); // @output {"field1":true,"field2":4345,"fieldA1":4534,"fieldA2":false}
+    printResult({...a, field1: true, field2: 4345}); // @output {"fieldA1":4534,"fieldA2":false,"field1":true,"field2":4345}
+    printResult({field1: true, ...a, field2: 4345}); // @output {"field1":true,"fieldA1":4534,"fieldA2":false,"field2":4345}
+
+    record {|int fieldB1; error fieldB2;|} b = {fieldB1:454, fieldB2: error("NPE")};
+
+    printResult({field1: true, field2: 4345, ...b}); // @output {"field1":true,"field2":4345,"fieldB1":454,"fieldB2":error("NPE")}
+    printResult({...b, field1: true, field2: 4345}); // @output {"fieldB1":454,"fieldB2":error("NPE"),"field1":true,"field2":4345}
+    printResult({field1: true, ...b, field2: 4345}); // @output {"field1":true,"fieldB1":454,"fieldB2":error("NPE"),"field2":4345}
+
+    printResult({...a, field2: 4345, ...b}); // @output {"fieldA1":4534,"fieldA2":false,"field2":4345,"fieldB1":454,"fieldB2":error("NPE")}
+    printResult({...a, ...b}); // @output {"fieldA1":4534,"fieldA2":false,"fieldB1":454,"fieldB2":error("NPE")}
+    printResult({field1: true, ...b, field2: 4345, ...a}); // @output {"field1":true,"fieldB1":454,"fieldB2":error("NPE"),"field2":4345,"fieldA1":4534,"fieldA2":false}
+}
+
+function printResult(map<any|error> result) {
+    io:println(result);
+}
+
+Test-Case: output
+Description: Test usage of spread field in mapping constructor using open record types.
+Labels: any, mapping-constructor-expr, map-type, record-rest-descriptor, record-type, spread-field, 
+       string
+
+function init() {
+    map<string> c = {fName: "John", lName: "Wick"};
+    printResult({...c}); // @output {"fName":"John","lName":"Wick"}
+
+    record {int fieldA1; boolean fieldA2;} d = {fieldA1: 4534, fieldA2: false};
+    printResult({...d}); // @output {"fieldA1":4534,"fieldA2":false}
+
+    record {|int fieldA1; boolean...; |} e = {fieldA1: 4534, "fieldA2": false};
+    printResult({...d}); // @output {"fieldA1":4534,"fieldA2":false}
+}
+
+function printResult(map<any> result) {
+    io:println(result);
+}
+
+Test-Case: output
+Description: Test mapping value of spread field contains spread field.
```

Comment:
Why do we need such a test? `b` can be anything, right?

Reply:
Yes, `b` can be any mapping value. The purpose of this test is to test the mapping value of the spread field containing another spread field inside.